mlflow/data/evaluation_dataset.py

```diff
@@ -130,11 +146,17 @@ def _hash_array_like_element_as_bytes(v):
             except TypeError:
                 return b""  # Skip unhashable types by returning an empty byte string
 
-        if Version(pd.__version__) >= Version("2.1.0"):
-            data = data.map(_hash_array_like_element_as_bytes)
+        if is_polars_dataframe(data):
+            return polars_to_bytes(data)
+        elif is_pyarrow_table(data):
+            return pyarrow_to_bytes(data)
         else:
-            data = data.applymap(_hash_array_like_element_as_bytes)
+            if Version(pd.__version__) >= Version("2.1.0"):
+                data = data.map(_hash_array_like_element_as_bytes)
+            else:
+                data = data.applymap(_hash_array_like_element_as_bytes)
         return _hash_uint64_ndarray_as_bytes(pd.util.hash_pandas_object(data))
+
```

Comment:
Not _too_ happy with this. I couldn't squeeze polars and pyarrow into the same behaviour of pandas without converting